client/components/chat-list-section.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import NewChatForm from './new-chat-form';
+import ChatList from './chat-list';
+
+export default class ChatListSection extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      formIsOpen: false,
+      form: {
+        chatName: '',
+        userName: ''
+      },
+      chatRooms: []
+    };
+    this.openNewChatForm = this.openNewChatForm.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.closeForm = this.closeForm.bind(this);
+    this.submitForm = this.submitForm.bind(this);
+    this.buildNewState = this.buildNewState.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllChatRooms();
+  }
+
+  openNewChatForm() {
+    const newState = this.buildNewState();
+    newState.formIsOpen = true;
+    this.setState(newState);
+  }
+
+  handleChange(target) {
+    const newState = this.buildNewState();
+    newState.form = Object.assign({}, this.state.form);
+    newState.formIsOpen = true;
+    if (target.id === 'chat-name') {
+      newState.form.chatName = target.value;
+      this.setState(newState);
+      return;
+    }
+    newState.form.userName = target.value;
+    this.setState(newState);
+  }
+
+  submitForm() {
+    const init = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(this.state.form)
+    };
+    fetch('/api/newRoom', init)
+      .then(response => response.json())
+      .then(result => {
+        this.getAllChatRooms();
+      })
```

Comment:
The server should respond with the created chat room. Instead of re-downloading the whole world, just add what you got back to what you already have.